runtime/android/core/src/org/xwalk/core/XWalkUpdater.java

```diff
@@ -148,42 +245,50 @@ public void run() {
      *         doesn't need to be updated, or the Crosswalk runtime has not been initialized yet.
      */
     public boolean updateXWalkRuntime() {
-        if (mDialogManager.isShowingDialog()) return false;
+        if (mIsDownloading || mDialogManager.isShowingDialog()) return false;
 
         int status = XWalkLibraryLoader.getLibraryStatus();
         if (status == XWalkLibraryInterface.STATUS_PENDING ||
                 status == XWalkLibraryInterface.STATUS_MATCH) return false;
 
-        Log.d(TAG, "Update the Crosswalk runtime with status " + status);
-        mDialogManager.showInitializationError(status, mCancelCommand, mDownloadCommand);
-        return true;
-    }
-
-    /**
-     * Dismiss the dialog showing and waiting for user's input.
-     *
-     * @return Return false if no dialog is being displayed, true if dismissed the showing dialog.
-     */
-    public boolean dismissDialog() {
-        if (!mDialogManager.isShowingDialog()) return false;
-        mDialogManager.dismissDialog();
+        if (isDownloadMode()) {
```

Comment:
In shared mode, the developer doesn't have to implement the callback method `onXWalkUpdateFailed()`, because there will be an alert dialog. But in download mode, the developer must handle the error. Which means that we can't guarantee that same code can apply to all modes. So it doesn't make sense to provide only one interface `updateXWalkRuntime`. We can provide another interface `updateXWalkRuntimeSilently` for the download mode